package/kedro_viz/models/graph.py

```diff
@@ -497,10 +508,57 @@ def __post_init__(self, data_node: DataNode):
             with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
                 self.plot = json.load(fs_file)
 
+        if data_node.is_metric_node():
+            if not dataset._exists() or self.filepath is None:
+                return
+
+            import pandas as pd
+            from kedro.extras.datasets.tracking.metrics_dataset import MetricsDataSet
+
+            dataset = cast(MetricsDataSet, dataset)
+            load_path = get_filepath_str(dataset._get_load_path(), dataset._protocol)
+            with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
+                self.metrics = json.load(fs_file)
+            metrics_data = data_loader.load_data_for_all_versions(self.filepath)

```

Comment:
I'm a bit concerned that we call the API `load_data_for_all_versions` but it has a hidden implicit limit of 10. I'm also concerned that the underlying implementation is limited to just JSON but the interface is generic. IMO we can improve the interface this way:

```python
data_loader.load_data_for_multiple_versions(dataset=dataset, num_versions=10)
```

Then all of the path wranging will happen inside `load_data_for_multiple_versions` as well as the load operation will use the native `dataset.load`. For each version, you can instantiate a new dataset object and load it this way:

```python
dataset_class = type(dataset)
for version in versions:
    dataset_instance = dataset_class(filepath=filepath, version=version)
    data = dataset_instance.load()
```

Something like this. Let me know if you want to pair on this.